pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -475,6 +530,86 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *api.Ser
 		status.Ingress = append(status.Ingress, api.LoadBalancerIngress{IP: floatIP.FloatingIP})
 	}
 
+	if lbaas.opts.ManageSecurityGroups {
+
+		lbSecGroupCreateOpts := groups.CreateOpts{
+			Name:        getSecurityGroupName(clusterName, apiService),
+			Description: fmt.Sprintf("Securty Group for %v Service LoadBalancer", apiService.Name),
+		}
+
+		lbSecGroup, err := groups.Create(lbaas.network, lbSecGroupCreateOpts).Extract()
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, port := range ports {
+
+			for _, sourceRange := range sourceRanges.StringSlice() {
+				lbSecGroupRuleCreateOpts := rules.CreateOpts{
+					Direction:      "ingress",
+					PortRangeMax:   int(port.Port),
+					PortRangeMin:   int(port.Port),
+					Protocol:       strings.ToLower(string(port.Protocol)),
+					RemoteIPPrefix: string(sourceRange),
+					SecGroupID:     lbSecGroup.ID,
+					EtherType:      "IPv4",
+				}
+
+				_, err := rules.Create(lbaas.network, lbSecGroupRuleCreateOpts).Extract()
+
+				if err != nil {
+					return nil, err
+				}
+
+				lbaas.createNodeSecurityGroup(int(port.Port), string(port.Protocol), lbSecGroup.ID)
```

Comment:
What this does is create the rules for traffic from the loadbalancer to the node.

What the `rules.Create()` does is allow traffic from outside to the loadbalancer.

There has to be 2 sets of rules, as the ports for each side are different.